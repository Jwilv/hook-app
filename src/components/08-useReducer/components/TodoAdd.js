import React from 'react'
import { useForm } from '../../../hooks/useForm'

export const TodoAdd = ({handleAddTodo}) => {

    const [{ description }, handleInputChanGet, reset] = useForm({
        description: ''
    })

    const handleSubmit = (event) => {
        event.preventDefault();

        if (description.trim().length <= 1) return;

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        }

        handleAddTodo(newTodo);
        reset();
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                name='description'
                placeholder='Incluir...'
                autoComplete='off'
                className='form-control'
                value={description}
                onChange={handleInputChanGet}
            />
            <button
                type='submit'
                className='btn btn-block btn-outline-primary mt-1 btn-block'
            >
                Agregar
            </button>
        </form>
    )
}
