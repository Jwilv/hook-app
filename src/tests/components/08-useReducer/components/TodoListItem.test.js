import React from 'react';
import { shallow } from 'enzyme'
import { TodoListItem } from '../../../../components/08-useReducer/components/TodoListItem';
import { demoTodo } from '../../../fixtures/demoTodos';


describe('test em el componente <TodoListItem />', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();


    const wrapper = shallow(<TodoListItem 
        todo={demoTodo[0]}
        i={0}
        handleToggle={handleToggle}
        handleDelete={handleDelete}
        />)


    test('debe de mostrarse corectamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de llamar la funcion handleDelete', () => {
        
        wrapper.find('button').simulate('click');
        expect( handleDelete ).toHaveBeenCalledWith(demoTodo[0].id);

    });

    test('debe de llamar la funcion handleToggle', () => {
        
        wrapper.find('p').simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith(demoTodo[0].id);

    });

    test('debe de mostrar el texto corectamente', () => {

        expect(wrapper.find('p').text()).toBe(`1. ${demoTodo[0].desc}`)
    });
})