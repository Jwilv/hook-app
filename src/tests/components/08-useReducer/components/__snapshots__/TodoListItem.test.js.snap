// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test em el componente <TodoListItem /> debe de mostrarse corectamente  1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    aprender a jugar
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
